### PR TITLE
feat: harbor backup

### PR DESCRIPTION
**Pull request by srodenhuis**
This PR brings:

- Backup the Harbor registry DB (using pg_dump) and store the backup in a Minio bucket
- A generic backup script that can be used in other jobs to create backups of postgres dbs and store the backup in a Minio bucket

This feature is now separated from the Harbor upgrade PR.

## Checklist

- [ ] Create new tools image
- [ ] Test backup

### CODE REVIEW

File: tools/Dockerfile [hunk 1]
```diff
@@ -53,8 +53,8 @@ RUN apt-get update -qq \
   vim \
   nano \
   groff \
-  awscli \
   rsync \
+  postgresql-client \
```
Comment (j-zimnowoda) [inline tools/Dockerfile:57]: Would it be possible to use `bitnami/postgresql` that contains all required tooling?

Comment (j-zimnowoda) [reply tools/Dockerfile:57]: Ah I see it is noot that easy as the aws CLI is also needed

File: helmfile.d/helmfile-99.post.yaml [hunk 2]
```diff
@@ -30,6 +30,12 @@ releases:
       pkg: harbor
       group: jobs
     <<: *jobs
+  - name: job-harbor-backup
+    installed: {{ $a | get "harbor.backup.enabled" }}
```
Comment (j-zimnowoda) [inline helmfile.d/helmfile-99.post.yaml:34]: I believe it should be:
```
installed: {{ and ($a.harbor.enabled) ($a | get "harbor.backup.enabled"  false) }}
```

You could also add the `harbor.backup.enabled` to the `defaults.yaml` file then you do not need to use `get`

File: values/harbor/harbor.gotmpl [hunk 3]
```diff
@@ -6,13 +6,13 @@
...
 {{- $notarySecretName := ($notaryDomain | replace "." "-") }}
 {{- $externalUrl := printf "https://%s" $harborDomain }}
-{{- $tag := $h | get "image.tag" "v2.4.0" }}
+{{- $tag := $h | get "image.tag" "v2.5.4" }}
 
 externalURL: {{ $externalUrl }}
 fullnameOverride: harbor
 # logLevel - debug, info, warning, error or fatal
 logLevel: warning
-harborAdminPassword: {{ $h | get "adminPassword" $v.otomi.adminPassword }}
```
Comment (j-zimnowoda) [inline values/harbor/harbor.gotmpl:15]: I remember that you mentioned that there was an issue with logic here causing harbor db password being taken from the Helm chart. Still, I am wondering if this change can have any side effects on existing deployments. If so then we need to include it in the release notes.

File: values/harbor/harbor.gotmpl [hunk 4]
```diff
@@ -117,7 +117,7 @@ database:
         cpu: 50m
         memory: 256Mi
       {{- end }}
-  password: {{ $h | get "adminPassword" $v.otomi.adminPassword }}
+  password: {{ $v.otomi.adminPassword }}
```
Comment (j-zimnowoda) [inline values/harbor/harbor.gotmpl:120]: same as above

File: values/jobs/scripts/backup-db.sh [hunk 5]
```diff
@@ -0,0 +1,4 @@
+set -e
+set -o pipefail
+echo "MAINTENANCE TASK: creating backup of $PGDATABASE db and storing it in Minio $BUCKET bucket"
+pg_dump -O -Fc --no-owner | aws --endpoint-url $ENDPOINT s3 cp - s3://$BUCKET/$POSTFIX/backup_$(date +%Y%m%d%H%M%S%N).dump
```
Comment (j-zimnowoda) [inline values/jobs/scripts/backup-db.sh:4]: Could you rephrase the echo message, so it includes the filename of the backup?

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]

**Review by Morriz — CHANGES_REQUESTED** [review]
Looking good. Thanks for moving it to our values/jobs.

Minor remark left

File: tools/Dockerfile [hunk 6]
```diff
@@ -53,7 +53,6 @@ RUN apt-get update -qq \
   vim \
   nano \
   groff \
-  awscli \
```
[the remark this answers is not in the document]
Comment (j-zimnowoda) [reply tools/Dockerfile:56]: in line 38 the was cli is already installed

File: values/harbor/harbor.gotmpl (hunk 4 above)
[the remark this answers is not in the document]
Comment (srodenhuis) [reply values/harbor/harbor.gotmpl:120]: Indeed. we'll fix it

**Comment by staticvoid255** [comment]
Am I correct in the assumptions that this is dependent on #945?

**Review by j-zimnowoda — APPROVED** [review]
